Update gradle to 3.3.0, remove buildToolsVersion

hendercine opened the pull request:
Gradle was simply not syncing on Android Studio 3.3 until I updated the Gradle Version, the Android Plugin Version and refactored the modules to remove old buildToolsVersions. This pull requests updates the above mentioned components.

I also had to comment out/remove the externalBuildDir code (lines 41 - 46 in the project level build.gradle file) in order for Android Studio to run the project, however, I am not able to debug that at this time.

Commit:
Update gradle to 3.3.0, remove buildToolsVersion

hendercine commented:
Perhaps the problem is that I used my work chrome account to submit the pull request?

ojw28 commented:
I'm not sure about this change. I've put some questions below. Note that upgrading also introduces a warning due to [this issue](https://github.com/novoda/bintray-release/issues/260).

> Gradle was simply not syncing on Android Studio 3.3 until I updated the Gradle Version, the Android Plugin Version and refactored the modules to remove old buildToolsVersions. This pull requests updates the above mentioned components.

* The current `dev-v2` branch syncs fine in AS 3.3 for me. So it's unclear whether this was just an issue with your setup, rather than a wider issue that needs fixing.
* Why are you removing `buildToolsVersion` everywhere? Shouldn't you just update the version used to `28.0.3` in `constants.gradle`?

hendercine commented:
Hi Oliver (@ojw28 ),

Thanks for looking into this. To address your questions:

> I'm not sure about this change. I've put some questions below. Note that upgrading also introduces a warning due to [this issue](https://github.com/novoda/bintray-release/issues/260).

I've been seeing this warning in other projects relating to the variantOutput API and have yet to manage to debug it. It seems, however, that this warning has been occurring for a range of tasks that the community seems to think is a bug in the IDE, see: [this](https://stackoverflow.com/questions/52470044/warning-api-variant-getjavacompile-is-obsolete-and-has-been-replaced-with) and [this](stackoverflow.com/a/54213973/304270).

> > Gradle was simply not syncing on Android Studio 3.3 until I updated the Gradle Version, the Android Plugin Version and refactored the modules to remove old buildToolsVersions. This pull requests updates the above mentioned components.
> 
> * The current `dev-v2` branch syncs fine in AS 3.3 for me. So it's unclear whether this was just an issue with your setup, rather than a wider issue that needs fixing.

I was actually having trouble compiling 'release-v2' and modified it with the included changes before deciding to attempt a pull-request and learning the contribution guidelines - i.e. sending pull requests to 'dev-v2'. Perhaps, there are changes in 'dev-v2' affecting this that haven't been merged to 'release-v2' yet? Of course, it's entirely possible that it's an issue with my setup. The decision to migrate to the newest Gradle Plugin was based on information provided in the errors and what I thought was generally considered to be best practice, but you're totally right that it might not represent a wider issue.

> * Why are you removing `buildToolsVersion` everywhere? Shouldn't you just update the version used to `28.0.3` in `constants.gradle`?

When one tries to build with Gradle Plugin 3.3.0 with a buildToolsVersion < 28.0.3 gradle produces this warning and recommended action:

`WARNING: The specified Android SDK Build Tools version (27.0.3) is ignored, as it is below the minimum supported version (28.0.3) for Android Gradle Plugin 3.3.0.
Android SDK Build Tools 28.0.3 will be used.
To suppress this warning, remove "buildToolsVersion '27.0.3'" from your build.gradle file, as each version of the Android Gradle Plugin now has a default version of the build tools.
Remove Build Tools version and sync project`

I read that as stating that specifying a buildToolsVersion is now redundant in Gradle 3.3.0+. The IDE automatically performed the removal of buildToolsVersion everywhere.

ojw28 commented:
Could you check to see whether `dev-v2` works fine in your setup without modification? If so, I think it's OK to leave things as is for now. If not, we'll look at what the minimum is that we need to do to get to a working state. Thanks!

Commit:
feat($audio): add device output control

ojw28 commented:
We will push a change to upgrade gradle shortly. It'll be easier for us to commit a change directly (this pull request also seems to have picked up an unrelated change to do with output control!). Expect the change to land in `dev-v2` sometime next week. Thanks!

hendercine commented:
Great! Thanks for the update, Oliver! You guys are so on top of it. Super humbled and impressed.

Best,
James Henderson

> On Feb 8, 2019, at 1:06 PM, Oliver Woodman <notifications@github.com> wrote:
> 
> We will push a change to upgrade gradle shortly. It'll be easier for us to commit a change directly (this pull request also seems to have picked up an unrelated change to do with output control!). Expect the change to land in dev-v2 sometime next week. Thanks!
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub <https://github.com/google/ExoPlayer/pull/5439#issuecomment-461946736>, or mute the thread <https://github.com/notifications/unsubscribe-auth/ARqHDOVtZ-nMHCYotGgy2cs4uYPHxv6gks5vLebEgaJpZM4aTrVq>.
>
